src/sentry/replays/consumers/recording.py

```diff
@@ -39,17 +41,50 @@ class ProcessReplayRecordingStrategyFactory(ProcessingStrategyFactory[KafkaPaylo
     chunks.
     """
 
+    def __init__(
+        self,
+        input_block_size: int,
+        max_batch_size: int,
+        max_batch_time: int,
+        num_processes: int,
+        output_block_size: int,
+    ) -> None:
+        self.input_block_size = input_block_size
+        self.max_batch_size = max_batch_size
+        self.max_batch_time = max_batch_time
+        self.num_processes = num_processes
+        self.output_block_size = output_block_size
+        self.use_multi_proc = self.num_processes > 1
+
     def create_with_partitions(
         self,
         commit: Commit,
         partitions: Mapping[Partition, int],
     ) -> Any:
-        step = RunTaskInThreads(
-            processing_function=move_replay_to_permanent_storage,
-            concurrency=4,
-            max_pending_futures=50,
-            next_step=CommitOffsets(commit),
-        )
+        # Give mypy some help.
+        step: Union[
+            RunTaskWithMultiprocessing[MessageContext, Any], RunTaskInThreads[MessageContext, Any]
```

Comment:
can you use the common parent class `ProcessingStrategy[MessageContext, Any]` instead of a union?